File: easy/27/remove_element.ts
function removeElement(nums: number[], val: number): number {
  let qty: number = 0;
  let lng_idx: number = nums.length - 1;

  let i: number = 0;
  while (i <= lng_idx + 1) {
    if (nums[i] === val) {
      nums[i] = nums[lng_idx];
      nums[lng_idx] = null;

      lng_idx -= 1;
    } else {
      i += 1;
      qty += 1;
    }
  }

  return qty - 1;
}

File: easy/27/remove_element.go
package main

import "fmt"

func removeElement(nums []int, val int) int {
	k := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			nums[k] = nums[i]
			k++
		}
	}
	return k
}

func main() {
	nums := []int{3, 2, 2, 2, 3}
	val := 3
	k := removeElement(nums, val)

	// Expect {2, 2, 2, _, _}
	fmt.Println(nums)
	// Expect: 3
	fmt.Println(k)
}
